### DISCUSSION TITLE
Axios integration & English language

### DISCUSSION THREAD

**Discussion by Nadaabdalkader** [ant-design/pro-chat#89] [Q&A] [answered]
Hello all!

I have some questions actually. 

First one, can I use Axios responses in the component?
Secondly, is there a way to show the text in components in English? For example, the message input place holder and the delete confirmation message/buttons?

Additionally, can I hide the delete history button?

Finally, any plans to support state persistence for chat messages?

Thanks in advance!

**Comment by ONLY-yours** [comment, marked as answer]
@Nadaabdalkader sry for taking so long to reply to you

first question：u can use any network, components only care about text return 

second：placeholder can help u change  the message input place holder, about delete config buttons,u can use actions : render to modify it

```tsx
        actions={{
          render: (defaultDoms) => {
            return <></>
          }
        }}
```
> this can help u hide delete button

last: we do not suggest do persistence directly in the component, we think its should judged by developer, however pro-chat provide a doc about how to work with state persistence, u can read, maybe its will help u 
https://pro-chat.antdigital.dev/guide/storage

**Reply by Nadaabdalkader** [reply]
Thank you so much for answering all of my questions, it was really helpful :))
